### PR TITLE
Reset rework. Closes #121

### LINKED ISSUE #121: Some room traversals require a room reset
There are some links that may not be traversable without resetting the room by exiting and re-entering, in rooms where items can make it so a plausible path could do just that. In those cases, we need an indication that the player needs to exit and re-enter (so that the logic can tell if there's something problematic about doing that)

- Early Supers, from left to right, requires a reset at the left door
- Speed booster hall, in either direction, more or less requires a reset because of the crumble blocks. Otherwise, the heat frames requirements will be way off.
- Any other places?

### PR DESCRIPTION
**Pull request by DIProgan**
Adds several new runways and links that needs a "resetRoomAtNode" to work properly.

### CODE REVIEW

File: region/brinstar/green.json [hunk 1]
```diff
@@ -443,7 +443,8 @@
               "requires": [
                 {"or": [
                   "canDash",
-                  "canMockball"
+                  "canMockball",
+                  {"resetRoomAtNode": [1]}
```
Comment (fdtaillefer) [inline region/brinstar/green.json]: This needs to be outside the `or`

File: region/brinstar/red.json [hunk 2]
```diff
@@ -765,7 +765,7 @@
             "runways": [
               {
                 "length": 8,
-                "requires": null
+                "requires": [ {"resetRoomAtNode": [1, 2, 4, 5]} ]
```
Comment (fdtaillefer) [inline region/brinstar/red.json]: I think 5 is a bit convoluted with the green gate. I'd leave that out.

File: region/norfair/east.json [hunk 3]
```diff
@@ -1742,7 +1742,7 @@
                     "heatProof",
                     "canHeatRun"
                   ]},
-                  "reset",
+                  {"resetRoomAtNode": [1]},
```
Comment (fdtaillefer) [inline region/norfair/east.json]: Needs reset at opposite side.
Also, traversing from 1 to 2 requires a reset at 1.

File: region/norfair/east.json [hunk 4]
```diff
@@ -1778,7 +1778,7 @@
                     "heatProof",
                     "canHeatRun"
                   ]},
-                  "reset",
+                  {"resetRoomAtNode": [2]},
```
Comment (fdtaillefer) [inline region/norfair/east.json]: Needs reset at opposite side.
Also, traversing from 2 to 1 requires a reset at 2.

File: region/brinstar/green.json (hunk 1 above)
Comment (DIProgan) [reply region/brinstar/green.json]: Ops

**Pull request merged**